lib/json/pure/generator.rb

```diff
@@ -339,6 +339,7 @@ def generate(obj)
           private def fast_serialize_string(string, buf) # :nodoc:
             buf << '"'
             string = string.encode(::Encoding::UTF_8) unless string.encoding == ::Encoding::UTF_8
+            raise GeneratorError, "source sequence is illegal/malformed utf-8" unless string.valid_encoding?
```

Comment:
Probably need to do it in the Ruby 2.3 version too?

Reply:
That one calls `String#to_json` so it's fine